cmd/juju/application/bundle/bundle.go

```diff
@@ -315,3 +325,55 @@ func verifyBundle(data *charm.BundleData, bundleDir string) error {
 	}
 	return errors.Trace(verifyError)
 }
+
+func verifyMixedSeriesBasesMatch(data *charm.BundleData) error {
+	if data == nil {
+		return nil
+	}
+	if data.Series != "" && data.DefaultBase != "" {
+		b, err := corebase.ParseBaseFromString(data.DefaultBase)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		s, err := corebase.GetSeriesFromBase(b)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if s != data.Series {
+			return errors.NewNotValid(nil, fmt.Sprintf("bundle series %q and base %q must match if supplied", data.Series, data.DefaultBase))
```

Comment:
This isn't the same comment as in 3.3, are you missing commits in the back port?

Reply:
Yes, I think this was a change made to the PR (responding to comments) which took place after I had cherry-picked
Since I squashed, ass that should be required to fix any such problems is to drop the commit and re-cherry-pick it

Reply:
This has been done now. The resulting diff is:
https://pastebin.canonical.com/p/wMQjtnR5qH/